Additional comets for Celestia (2025)

DaveBowman2001 opened the pull request:
I have noticed that the last time ```comets.ssc``` was updated, it was over 16 years ago by Chris Laurel himself (https://github.com/CelestiaProject/CelestiaContent/commit/722fd82c1c9a9f93f8f24d2103ff17b942b6460d). But this only includes 4 comets:
- 1P/Halley
- 19P/Borrelly
- 153P/Ikeya–Zhang
- C/1995 O1 (Hale–Bopp)

This PR adds over 19 comets, mostly notable ones, to the main catalog for Celestia. These are the following:
- 2P/Encke
- 17P/Holmes
- 21P/Giacobini–Zinner
- 26P/Grigg–Skjellerup
- 29P/Schwassmann–Wachmann
- 46P/Wirtanen
- 55P/Tempel–Tuttle
- 96P/Machholz
- 109P/Swift–Tuttle
- 167P/CINEOS
- C/1956 R1 (Arend–Roland)
- C/1973 E1 (Kohoutek)
- C/1983 H1 (IRAS–Araki–Alcock)
- C/1996 B2 (Hyakutake)
- C/2006 P1 (McNaught)
- C/2013 A1 (Siding Spring)
- C/2014 UN271 (Bernardinelli–Bernstein)
- C/2020 F3 (NEOWISE)
- C/2023 A3 (Tsuchinshan–ATLAS)

Most of the comets added here have known physical properties, most notably nucleus diameters (and a few also have known rotation periods). Also contains updates for existing comets as well

DaveBowman2001 commented:
I intentionally left out the following comets below as they have existing custom models (available as addons) that I do not know what their creative licenses are:
- 67P/Churyumov–Gerasimenko
- 81P/Wild
- 103P/Hartley

DaveBowman2001 commented:
Slight change in the naming format for comets https://github.com/CelestiaProject/CelestiaContent/pull/229/commits/92dd004576332f3b6b81a9ae1469a161e720eb15 as discussed on Discord with @SpaceExplorer2008 and @pedro-fixingstuff 

Also added more comets:
- 29P/Schwassmann–Wachmann
- C/1983 H1 (IRAS–Araki–Alcock)
- C/2014 UN271 (Bernardinelli–Bernstein)

DaveBowman2001 commented:
New commit adds the following comets:
- 167P/CINEOS
- C/2006 P1 (McNaught)
- C/2013 A1 (Siding Spring)

DaveBowman2001 commented:
This should be the last of the comets to be added. The visited comets (9P, 67P, 81P, 103P) will be added at a later date when their custom 3D models are sorted out.

FINAL LIST OF NEW COMETS:
- 2P/Encke
- 17P/Holmes
- 21P/Giacobini-Zinner
- 26P/Grigg-Skjellerup
- 29P/Schwassmann-Wachmann
- 46P/Wirtanen
- 109P/Swift-Tuttle
- 167P/CINEOS
- C/1956 R1 (Arend-Roland)
- C/1973 E1 (Kohoutek)
- C/1983 H1 (IRAS-Araki-Alcock)
- C/1996 B2 (Hyakutake)
- C/2006 P1 (McNaught)
- C/2013 A1 (Siding Spring)
- C/2014 UN271 (Bernardinelli-Bernstein)
- C/2020 F3 (NEOWISE)
- C/2022 E3 (ZTF)
- C/2023 A3 (Tsuchinshan-ATLAS)
- C/2023 P1 (Nishimura)

TOTAL: 19

SevenSpheres commented:
Per discussion on Discord at least the following changes should be made:
- Remove C/2022 E3 & C/2023 P1, which are only significant due to being recent - they can be in addons
- Restore the established name format that replaces slashes with spaces, rather than hyphens (`1P Halley` not `1P-Halley`)

DaveBowman2001 commented:
> * Remove C/2022 E3 & C/2023 P1, which are only significant due to being recent - they can be in addons

They are now replaced with 55P/Tempel-Tuttle and 96P/Machholz https://github.com/CelestiaProject/CelestiaContent/pull/229/commits/1a066d3c62986435a55300e1ea91a0c7d90e4fe5

> * Restore the established name format that replaces slashes with spaces, rather than hyphens (`1P Halley` not `1P-Halley`)

OK fine, I removed the hyphens from the periodic/interstellar comets. But I'll keep the parentheses on non-periodic comets as they've been written with one on official catalogs anyway (including MPC and JPL-SBDB)

I still wish that we could somehow use the slashes when naming comets in Celestia. Their absence feels a little bit off imho, probably I'm used to seeing them on comet names all the time

SevenSpheres commented:
I'm not sure why they need to be replaced, the goal isn't to add as many comets as possible and personally I wouldn't include so many non-periodic comets in default... but now that each object has a reason for inclusion that isn't recentism I won't object to it.

(This will have to be approved by someone who can do so)

DaveBowman2001 commented:
> I'm not sure why they need to be replaced...

Originally, I wanted to add up to 21 comets, to bring up the total of 24 (including the already default comets). As for their replacements, 55P is the parent body of the Leonids and 96P is the only large sungrazing comet that is also a periodic one (in addition to being the parent of Marsden and Kracht sungrazers)

DaveBowman2001 commented:
@ajtribick, @375gnu this PR is ready for final review/merge

AstroChara reviewed (approved):
The file appears to contain all comets that is mentioned in the first post, and it doesn't break my Celestia install when I substitute current files with them, so I think it's probably good to go.
